Refactoring

egervari opened the pull request:
Lots of refactoring and performance enhancements

egervari committed:
Updated gitignore to include webstorm
Fixed a bunch of stuff in documentation
Fixed prettier script for windows and then ran prettier since it seemed to not be done
Refactored demos

egervari committed:
Refactored probes

egervari committed:
Added fix script and lint:fix
Renamed a lot of files
Simplified Event Dispatcher
Completely reworked reactions
Reworked probe classes to avoid instantiating so many functions

egervari committed:
Renamed files to include type of file
Reworked event emitters

egervari committed:
Refactoring execution-timeline

egervari committed:
optimized execution timeline to make it much faster

egervari committed:
Fixed performance issues with d3 graphs and fixed some bugs

egervari committed:
Refactored probe manager
Fixed global styles and refactored font families
Refactored timeline more

egervari committed:
wip

egervari committed:
more refactorings and performance improvements. updated packages

egervari committed:
Small change to timeline

egervari committed:
Refactored commands

egervari committed:
Adjusted package.json versions and bootstrap

egervari committed:
misc refactorings regarding commands, plugins, channels, etc.

egervari committed:
Fixed an issue where augury was initializing and dispatching events before the bootstrap code finished. I guess angular 7+ made things start up a lot faster so our code ran after probes were already firing. It's fixed now.

egervari committed:
Fixed memory leak with ScanHistoryReaction

andrewthauer commented on README.md:
````diff
@@ -57,7 +57,7 @@ _NOTE: The following assumes your application is a standard [Angular CLI](https:
    ```
 
-1. Create a `src/main.augury.ts` file in your application with the following content:
+2. Create a `src/main.augury.ts` file in your application with the following content:
````
Not a big deal, but the `1.` numbering was used intentional since markdown render's will automatically number them as per this convention.

egervari committed:
Added missing test so `yarn test` works

stevenkampen commented:
👏

andrewthauer commented on demos/performance-profiler/e2e/app.po.ts:
```diff
@@ -1,11 +1,11 @@
 
 export class AppPage {
-  navigateTo() {
+  public navigateTo() {
```
I like the fact that the TSLint is setup to enforce this convention. However, I typically don't bother to explicitly forcing the usage of public. That said, I do agree with only making things public that should be. I suppose the rationale here is to ensure the developer thinks about the scope for every single member?

egervari replied:
Exactly. I want access modifier to be intentional. I've been on too many projects where everything is declared public without consideration and so enforcing it makes sure we think about it. I think it makes the code look more aethestically pleasing and balanced too, seeing private/public as opposed to just one keyword. But I mostly appreciate the tools forcing one to think about it.

andrewthauer commented on demos/performance-profiler/karma.conf.js:
```diff
@@ -1,7 +1,7 @@
 // https://karma-runner.github.io/1.0/config/configuration-file.html
 
-module.exports = function (config) {
+module.exports = function(config) {
```
Do we even need this file? The rest of the repo is setup to use jest so would make most sense to use that. Thoughts?

egervari replied:
I never actually looked at this so I am not sure :) That space was put in there by prettier I think.

andrewthauer replied:
For sure, just making a note that I'd prefer to use `jest` over karma throughout.

santiago-elustondo commented:
This is amazing. 
Checking it out now. (It's a big PR, might take a couple days to go through it all)

andrewthauer commented on packages/core/src/commands/request-live-channel-command.class.ts:
```diff
@@ -0,0 +1,22 @@
+
+export interface RequestChannelCommandRequestParameters {
+  reducer: Reducer;
+  startFromEID?: number;
```
Sorry, what's `EID`?

egervari replied:
So... the past code had EID littered everywhere. I got stumped on this too. It means "eventId". You'll see that I did correct some of them but I left some alone since I figured I would get to it later (we're redoing channels anyway). I 100% agree with you - EID is so bizarre I think they all need to be removed.

andrewthauer commented on packages/core/src/framework/commands/command.class.ts:
```diff
@@ -0,0 +1,25 @@
+    );
+  }
+
+  public abstract parseReactions(results: ReactionResults): CommandResult;
```
These seems a little out of place on the base command class.

egervari replied:
I agree - I don't really want commands parsing reactions anyway, so we'll remove this in a future PR

andrewthauer commented on packages/core/src/framework/events/augury-event.ts:
```diff
@@ -1,31 +1,31 @@
 export interface ElapsedAuguryEvent extends ProcessedAuguryEvent {
-  auguryHandlingCompletionPerformanceStamp: number
-  auguryDrag: number
+  auguryHandlingCompletionPerformanceStamp: number;
```
Could this name just be `completedAt` or `completedAtTimestamp`?

egervari replied:
I think it could be. I don't like the name either (it was there from before). When I saw this, I was a bit scared to refactor it since there's no types but i can do this now.

andrewthauer commented on packages/core/src/framework/events/augury-event.ts:
```diff
@@ -1,31 +1,31 @@
-  auguryHandlingCompletionPerformanceStamp: number
-  auguryDrag: number
+  auguryHandlingCompletionPerformanceStamp: number;
+  auguryDrag: number;
```
Is this elapsed time? If so, perhaps renaming it might be more clear.

andrewthauer commented on packages/core/src/framework/plugins/augury-plugin-controller.class.ts:
```diff
@@ -0,0 +1,57 @@
+import { AuguryBridge, AuguryBridgeMessage, AuguryBridgeRequest } from './bridge';
+
+export class AuguryPluginController {
+  private readonly window: AuguryPluginWindow;
```
Is the intention that plugins always have windows? Ideally plugins (consumers) could be headless as well as GUI base (or even different intra process).

egervari replied:
There was no intention really. The existing code was using a window and I just refactored it to separate concerns a bit.

andrewthauer commented on packages/core/src/framework/plugins/bridge/augury-bridge-request.interface.ts:
```diff
@@ -0,0 +1,5 @@
+export interface AuguryBridgeRequest {
+  type: 'get_full_cd';
```
Wondering if this string/enum could be extract into an isolated type?

egervari replied:
Ooops, I missed this one. I did the message type, but not the request type.

andrewthauer commented on packages/core/src/framework/reactions/reaction-results.interface.ts:
```diff
@@ -1,8 +1,8 @@
-import { ReactionResult } from './reaction-result'
+import { ReactionResult } from './reaction-result.interface';
 
 // @todo: how to type this well?
```
Can this comment be removed now?

andrewthauer commented on packages/standalone-uis/execution-timeline/src/index.html:
```diff
@@ -1,59 +1,16 @@
-  <style>
-    @font-face {
-      font-family: 'RangleFont';
-      src: url('data:application/x-font-woff2;base64,d09GMgABAAAAATVoABIAAAAELgAAATT/AAEAQgAAAAAAAAAAAAAAAAAAAAAAAAAAG4KGFBytJBSHGAZgAJFKCIEKCYJzERAKistAifAAEoKteAE2AiQDnFALnFQABCAFhygHxDUMiQ1bsvCzB/5T5N6/tU62bbYIRDs5ZdA5xOT3CkCFFd1DnRDOCGvvIGbX8FboM1K029MBNjbtHVjPzTCqVhdZrf3///////////8vThahWztzyeylQiAQShBBRCkqIvLNNzF3Z9EtixYtzyxaUVZ1DvlgGLEZjS1ikbm1qpVNZPLxGKbYFTSfOZsvlv3KBDFSiUqhDw0ZrWNcwdjnjz ...
```
Shouldn't these be double quoted?

andrewthauer commented on packages/standalone-uis/execution-timeline/src/app/components/change-detection-details/change-detection-details.component.ts:
```diff
@@ -0,0 +1,71 @@
+  constructor(private bridge: BridgeService) {}
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    this.subscription = this.bridge.subscribe(message => {
```
Couldn't this cause a leak or unwanted side effects since it's on ngOnChanges?

egervari replied:
It could, but in this case it won't get called more than once - i verified. I'll change it to onInit - i think it should be the same thing. I think the old code was using the changes object probably which is why it was using on changes. It's not needed anymore though.

andrewthauer reviewed (approved):
Awesome work! A few comments, suggestions, but nothing sticking out as a major problem. Especially considering some additional things will change.

egervari committed:
change detection details component - using on init instead of on changes
renamed elapsed augury event timestamp from auguryHandlingCompletionPerformanceStamp to completedAtTimestamp

egervari committed:
Added augury bridge request type
